frontend/javascripts/oxalis/view/left-border-tabs/layer_settings_tab.tsx

```diff
@@ -357,17 +365,46 @@ class DatasetSettings extends React.PureComponent<DatasetSettingsProps, State> {
           }}
         >
           {volumeDescriptor != null ? (
-            <EditableTextLabel
-              margin="0 10px 0 0"
-              width={150}
-              value={readableName}
-              onChange={(newName) => {
-                this.props.onEditAnnotationLayer(volumeDescriptor.tracingId, {
-                  name: newName,
-                });
-              }}
-              label="Volume Layer Name"
-            />
+            <Tooltip
+              title={isReadableNameValidOrWarning === true ? null : isReadableNameValidOrWarning}
+            >
+              <span style={{ display: "inline-block" }}>
+                <EditableTextLabel
+                  margin="0 10px 0 0"
+                  width={150}
+                  value={readableName}
+                  isInvalid={isReadableNameValidOrWarning !== true}
+                  trimValue
+                  onChange={(newName) => {
+                    this.props.onEditAnnotationLayer(volumeDescriptor.tracingId, {
+                      name: newName,
+                    });
+                  }}
+                  rules={[
+                    {
+                      message: messages["tracing.volume_layer_name_duplication"],
+                      validator: (newLayerName) =>
+                        typeof newLayerName === "string"
+                          ? checkForLayerNameDuplication(
+                              newLayerName,
+                              dataset,
+                              tracing,
+                              readableName,
+                            ) === true
+                          : false,
+                    },
+                    {
+                      message: messages["tracing.volume_layer_name_includes_slash"],
+                      validator: (newLayerName) =>
```

Comment:
My idea was to do the filtering at the caller site, so before `validateReadableLayerName` is called. This way, the `nameNotToCount` argument wouldn't be needed and the function interface would be smaller and easier to understand. However, I see that this would need to be done at two spots, which is not ideal - so it's also fine as is in my opinion.